app/controllers/admin/administrators/administrators_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Admin
+  module Administrators
+    class AdministratorsController < Admin::BaseController
+      skip_after_action :verify_authorized, only: :index
+      skip_after_action :verify_policy_scoped, except: :index
+
+      def index
+        @administrators = policy_scope(User).admins
+      end
+
+      def new
+        authorize AdminProfile
+
+        if params[:continue]
+          @user = user_from_session
+        else
+          session.delete(:administrator_user)
+          @user = User.new
+        end
+
+        authorize @user
+      end
+
+      def create
+        authorize AdminProfile
+        @user = User.new(permitted_attributes(User))
+
+        if @user.invalid?
+          render :new and return
+        end
+
+        session[:administrator_user] = @user
+      end
+
+      def success
+        user = User.new(permitted_attributes(User))
+        user.confirm
+
+        authorize user, :create?
+        authorize AdminProfile, :create?
+
+        ActiveRecord::Base.transaction do
+          user.save!
+          AdminProfile.create!(user: user)
+        end
+        session.delete(:administrator_user)
+      end
```

Comment:
Not that it makes much of a practical difference, but naming is a bit surprising. 

I think `create` does stuff I would expect from a page named `confirm`, and `success` does things I would expect from `create`. 

As in, as of now, `create` shows preview of what you are about to do, and `success` actually creates the user and their profile, and shows a success message. 

I would expect `confirm` to show the preview, and `create` to create the user and show success message. 

But these are naming conventions, and what I wrote only sounds more reasonable in my head - definitely not an objective point.

Reply:
Yep, I'd agree with that

Reply:
I agree @kicferk1 but then the routes in these multi step workflows can be a bit tricky to name properly. That said, I think it's worth fixing this.  success -> create and create -> validate / check etc